sources/core/Stride.Core.Reflection/MemberPath.cs

```diff
@@ -357,17 +357,17 @@ public object GetValue(object rootObject)
         /// <param name="value">The returned value.</param>
         /// <returns><c>true</c> if evaluation of the path succeeded and the value is valid, <c>false</c> otherwise.</returns>
         /// <exception cref="System.ArgumentNullException">rootObject</exception>
-        public bool TryGetValue(object rootObject, out object value)
+        public bool TryGetValue(object rootObject, [MaybeNullWhen(false)] out object? value)
         {
-            if (rootObject == null) throw new ArgumentNullException(nameof(rootObject));
+            ArgumentNullException.ThrowIfNull(rootObject);
             value = null;
             try
             {
                 object nextObject = rootObject;
                 for (int i = 0; i < items.Count; i++)
                 {
                     var item = items[i];
-                    nextObject = item.GetValue(nextObject);
+                    nextObject = item.GetValue(nextObject)!;
```

Comment:
Wouldn't it be valid for this method to return both true *and* output null `value` at the same time ?
In which case, this `!` should be removed I think ?

Reply:
is removed